### PR TITLE
Avoid exclusive language

### LINKED ISSUE #2139: Avoid exclusive language `master`
There has already been discussion and agreement internally to no longer use `master` as the name of our default branch. This issue is to announce our intention publicly and hold us accountable for the change (since it's already been ~2 months from our last discussion).

TODO:
- Determine what processes are needed to make the change
- Make the change

### PR DESCRIPTION
**Pull request by dan2k3k4**
- Replace usage `master` with `main` for branch/environment documentation and tests
- Replace master/slave with primary/replica for database documentation
- Update official documentation links by replacing https://lagoon.readthedocs.io/ with https://docs.lagoon.sh/

Closes #2139

### CODE REVIEW

**Comment by dan2k3k4** [comment]
I ran [Mega-Linter](https://nvuillam.github.io/mega-linter) locally and I found the following dead links via the [markdown-lint-check](https://github.com/tcort/markdown-link-check) tool.

Some are false positives, e.g. localhost / anything with 503, some need to be adjusted to new documentation locations.

They are the following:

---

[✖] /images/2.gif → Status: 400
[✖] /images/3.png → Status: 400
[✖] /images/4.gif → Status: 400
[✖] /images/5.gif → Status: 400
[✖] /images/6.gif → Status: 400
[✖] /images/7.gif → Status: 400
[✖] /images/8.png → Status: 400
[✖] /images/step2_require.gif → Status: 400
[✖] /services/auth-server/ → Status: 400
[✖] /services/backup-handler/ → Status: 400
[✖] /services/drush-alias/ → Status: 400
[✖] /services/logs-db-ui/ → Status: 400
[✖] /services/logs-db/ → Status: 400
[✖] /services/logs2email/ → Status: 400
[✖] /services/logs2logs-db/ → Status: 400
[✖] /services/logs2microsoftteams/ → Status: 400
[✖] /services/logs2rocketchat/ → Status: 400
[✖] /services/logs2slack/ → Status: 400
[✖] /services/ssh/ → Status: 400
[✖] /services/ui/ → Status: 400
[✖] /services/webhook-handler/ → Status: 400
[✖] /services/webhooks2tasks/ → Status: 400
[✖] graphql_api.md → Status: 400
[✖] harbor-container-settings/harbor-core.md → Status: 400
[✖] harbor-container-settings/harbor-database.md → Status: 400
[✖] harbor-container-settings/harbor-jobservice.md → Status: 400
[✖] harbor-container-settings/harbor-trivy.md → Status: 400
[✖] harbor-container-settings/harborregistry.md → Status: 400
[✖] harbor-container-settings/harborregistryctl.md → Status: 400

---

[✖] http://php.net/manual/en/apc.configuration.php#ini.apc.enabled → Status: 404
[✖] http://php.net/manual/en/apc.configuration.php#ini.apc.shm-size → Status: 404

[✖] http://www.example.com/blog → Status: 404
[✖] http://www.example.com/blog\ → Status: 404
[✖] http://www.example.com\ → Status: 400

[✖] https://api.lagoon.amazeeio.cloud/graphql → Status: 401
[✖] https://hooks.lagoon.amazeeio.cloud → Status: 400

[✖] http://lagoon.au.amazee.io/ → Status: 503
[✖] http://lagoon.ch.amazee.io/ → Status: 503
[✖] http://lagoon.us.amazee.io/ → Status: 503

[✖] http://localhost:8888/ → Status: 0
[✖] http://localhost/ → Status: 0
[✖] https://localhost:8084/ → Status: 0

[✖] http://drupal.docker.amazee.io → Status: 0
[✖] http://nginx-drupal-example.docker.amazee.io → Status: 0

[✖] http://rest2tasks-lagoon-develop.192.168.42.100.xip.io/ → Status: 0
[✖] https://192.168.42.100:8443/console/project/lagoon-develop/browse/builds/lagoon?tab=history → Status: 0

[✖] https://docs.lagoon.sh/ → Status: 0

[✖] https://docs.openshift.org/latest/cli_reference/get_started_cli.html#cli-reference-get-started-cli → Status: 0
[✖] https://docs.openshift.org/latest/dev_guide/service_accounts.html → Status: 0

[✖] https://github.com/amazeeio/drupal-example-simple/blob/8.x/TESTING.md → Status: 404

[✖] https://github.com/amazeeio/lagoon/blob/docs/images/mariadb/mysql-backup.sh → Status: 404
[✖] https://github.com/amazeeio/lagoon/blob/docs/images/mariadb/mysql-backup.sh#L24 → Status: 404
[✖] https://github.com/amazeeio/lagoon/blob/main/docs/administering_lagoon/create-project.gql → Status: 404
[✖] https://github.com/amazeeio/lagoon/tree/0ea0472705f825feb5fe0b946a8f4c9fc0bba707/docs/docker-images/redis.md → Status: 404
[✖] https://github.com/amazeeio/lagoon/tree/5cd57792f638b841dca84d99b7fcbf06af793817/docs/using-lagoon-advanced/environment_types.md#environment-types → Status: 404
[✖] https://github.com/amazeeio/lagoon/tree/main/docs/using_lagoon/drupal → Status: 404

[✖] https://logs-db-ui-lagoon-main.ch.amazee.io/ → Status: 503

---

File: docs/administering-lagoon/installing-lagoon-on-openshift.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Lagoon is not only capable of deploying into OpenShift, it actually runs in Open
 
 Luckily we can use the local development environment to kickstart another Lagoon in any OpenShift, running somewhere in the world.
 
-Check the [Requirements for OpenShift by Lagoon](https://github.com/AlannaBurke/lagoon/tree/9331f688073fb0d34541bf3fb5fe65d2a2c665aa/docs/administering-lagoon/openshift_requirements.md) before continuing.
```
Comment (AlannaBurke) [inline docs/administering-lagoon/installing-lagoon-on-openshift.md:7]: you have NO IDEA how many times i searched and fixed these links. how did this get here???

File: docs/contributing-to-lagoon/contributing.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 # Contributing
 
-We gladly welcome any and all contributions to Lagoon! 
+We gladly welcome any and all contributions to Lagoon!
 
-### What kind of contributions do we need? Examples!
```
Comment (AlannaBurke) [inline docs/contributing-to-lagoon/contributing.md:5]: This is a stupid rule, I want that exclamation point there and I'm going to put it back.

File: docs/drupal/first-deployment-of-drupal.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ If all is correct, you will see a notification in your configured chat system \(
 
 ![Slack notification of a deployment starting.](../.gitbook/assets/first_deployment_slack_start.jpg)
 
-This tells you that Lagoon has just started to deploy your code. Depending on the size of the codebase and amount of containers, this will take a couple of seconds. Just relax. If you'd like to know what's happening now, check out the[ Build and Deploy Process of Lagoon](../using-lagoon-the-basics/build-and-deploy-process.md).
+This tells you that Lagoon has just started to deploy your code. Depending on the size of the codebase and amount of containers, this will take a couple of seconds. Just relax. If you'd like to know what's happening now, check out the[Build and Deploy Process of Lagoon](../using-lagoon-the-basics/build-and-deploy-process.md).
```
Comment (AlannaBurke) [inline docs/drupal/first-deployment-of-drupal.md]: A needed space got dropped out when the linter took the space out of the brackets here.

File: docs/drupal/first-deployment-of-drupal.md [hunk 4]
```diff
@@ -124,7 +124,7 @@ Do you really want to continue? (y/n):
 
 The reason for that is that the Drupal cannot resolve the path of the files directory. This most probably has to do that the Drupal is not fully configured or has a missing database. For a workaround you can use `drush rsync @self:sites/default/files @develop:sites/default/files`, but we suggest that you actually check your local and remote Drupal \(you can test with `drush status` to see if the files directory is correctly configured\).
 
-## 7. It's done!
+## 7. It's done
```
Comment (AlannaBurke) [inline docs/drupal/first-deployment-of-drupal.md]: Is there a way to tell the linter to ignore things? I know it's little, but i care about the tone of our docs and i want the occasional exclamation point in my headers.

File: docs/resources/glossary.md [hunk 5]
```diff
@@ -64,7 +64,7 @@ description: >-
 | Symfony |  |
 | TCP | Transmission Control Protocol, a standard that defines how to establish and maintain a network conversation through which application programs can exchange data. |
 | TLS | Transport Layer Security |
-| Trivy |  |
+| Trivy | A simple and comprehensive vulnerability scanner for containers, suitable for CI (Continuous Integration) |
```
Comment (AlannaBurke) [inline docs/resources/glossary.md]: Thanks for this, sometimes I toss a word into the glossary and forget to go grab the definition from my glossary (which is here, btw: https://docs.google.com/spreadsheets/d/1VX1URCt7sre1GsmYCtmLdy-IUcMZMfaL03zaRWgOpRY/edit#gid=0)

File: docs/using-lagoon-the-basics/local-development-environments.md [hunk 6]
```diff
@@ -17,7 +17,7 @@ Lagoon has traditionally worked best with `pygmy` , which is the amazee.io flavo
 
 `pygmy` is a [Ruby](https://www.ruby-lang.org/en/) gem, so to install it, run: `gem install pygmy`.  For detailed usage info on pygmy, see its [documentation](https://pygmy.readthedocs.io/).
 
-As announced in our[ blog post](https://www.amazee.io/blog/post/announcing-lando-integration-for-lagoon), Lagoon is now also compatible with Lando!  For more information, please see the documentation at [https://docs.lando.dev/config/lagoon.html](https://docs.lando.dev/config/lagoon.html) to get yourself up and running.
+As announced in our[blog post](https://www.amazee.io/blog/post/announcing-lando-integration-for-lagoon), Lagoon is now also compatible with Lando!  For more information, please see the documentation at [https://docs.lando.dev/config/lagoon.html](https://docs.lando.dev/config/lagoon.html) to get yourself up and running.
```
Comment (AlannaBurke) [inline docs/using-lagoon-the-basics/local-development-environments.md]: Needs deleted space back.

File: helpers/check_acme_routes.sh [hunk 7]
```diff
@@ -230,14 +230,14 @@ function notify_customer() {
...
+	MESSAGE="Your $ROUTE_HOSTNAME route is configured in the \`.lagoon.yml\` file to issue an TLS certificate from Lets Encrypt. Unfortunately Lagoon is unable to issue a certificate as $DNS_ERROR.\nTo be issued correctly, the DNS records for $ROUTE_HOSTNAME should point to $CLUSTER_HOSTNAME with an CNAME record (preferred) or to ${CLUSTER_IPS[*]} via an A record (also possible but not preferred).\nIf you don't need the SSL certificate or you are using a CDN that provides you with an TLS certificate, please update your .lagoon.yml file by setting the tls-acme parameter to false for $ROUTE_HOSTNAME, as described here:  https://docs.lagoon.sh/using-lagoon-the-basics/lagoon-yml#ssl-configuration-tls-acme.\nWe have now administratively disabled the issuing of Lets Encrypt certificate for $ROUTE_HOSTNAME in order to protect the cluster, this will be reset during the next deployment, therefore we suggest to resolve this issue as soon as possible. Feel free to reach out to us for further information.\nThanks you.\namazee.io team"
```
Comment (AlannaBurke) [inline helpers/check_acme_routes.sh]: Can we update the instances of `Lets Encrypt` to be correctly spelled as `Let's Encrypt`?

File: tests/files/drupal8-mariadb-single/web/sites/default/default.settings.php [hunk 8]
```diff
@@ -119,7 +119,7 @@
  * That is useful for primary/replica replication, as Drupal may try to connect
  * to a replica server when appropriate and if one is not available will simply
  * fall back to the single primary server (The terms primary/replica are
```
Comment (AlannaBurke) [inline tests/files/drupal8-mariadb-single/web/sites/default/default.settings.php]: I think we can remove that parenthetical now :)

File: tests/files/drupal8-mariadb/web/sites/default/default.settings.php [hunk 9]
```diff
@@ -119,7 +119,7 @@
  * That is useful for primary/replica replication, as Drupal may try to connect
  * to a replica server when appropriate and if one is not available will simply
  * fall back to the single primary server (The terms primary/replica are
- * traditionally referred to as master/slave in database server documentation).
+ * traditionally referred to as primary/replica in database server documentation).
```
Comment (AlannaBurke) [inline tests/files/drupal8-mariadb/web/sites/default/default.settings.php]: Same, i think we can remove this parenthetical.

File: tests/files/drupal8-postgres-single/web/sites/default/default.settings.php [hunk 10]
```diff
@@ -119,7 +119,7 @@
  * That is useful for primary/replica replication, as Drupal may try to connect
  * to a replica server when appropriate and if one is not available will simply
  * fall back to the single primary server (The terms primary/replica are
- * traditionally referred to as master/slave in database server documentation).
+ * traditionally referred to as primary/replica in database server documentation).
```
Comment (AlannaBurke) [inline tests/files/drupal8-postgres-single/web/sites/default/default.settings.php]: Same as above.

File: tests/files/drupal8-postgres/web/sites/default/default.settings.php [hunk 11]
```diff
@@ -119,7 +119,7 @@
  * That is useful for primary/replica replication, as Drupal may try to connect
  * to a replica server when appropriate and if one is not available will simply
  * fall back to the single primary server (The terms primary/replica are
- * traditionally referred to as master/slave in database server documentation).
+ * traditionally referred to as primary/replica in database server documentation).
```
Comment (AlannaBurke) [inline tests/files/drupal8-postgres/web/sites/default/default.settings.php]: Ok maybe I'll make a note on d.o about this.

File: tests/files/drupal9-mariadb/web/sites/default/default.settings.php [hunk 12]
```diff
@@ -129,7 +129,7 @@
  * That is useful for primary/replica replication, as Drupal may try to connect
  * to a replica server when appropriate and if one is not available will simply
  * fall back to the single primary server (The terms primary/replica are
- * traditionally referred to as master/slave in database server documentation).
+ * traditionally referred to as primary/replica in database server documentation).
```
Comment (AlannaBurke) [inline tests/files/drupal9-mariadb/web/sites/default/default.settings.php]: Agaiiin.

**Review by AlannaBurke — CHANGES_REQUESTED** [review]
Left a few comments. Docs-wise, this looks good to me, as long as those links are good, but I need to check through them all anyway. Obviously I can't speak to the changes made to the technical side of things.

**Comment by tobybellwood** [comment]
yeah - we'd need to deep-dive into the changes to Lagoon logic to assess this - maybe scoping this PR to docs/examples as a first effort will help it get over the line?

File: docs/contributing-to-lagoon/contributing.md (hunk 2 above)
Comment (dan2k3k4) [reply docs/contributing-to-lagoon/contributing.md:5]: I think the linting I applied removes ending punctuation for titles, I can revert those

File: docs/drupal/first-deployment-of-drupal.md (hunk 3 above)
Comment (dan2k3k4) [reply docs/drupal/first-deployment-of-drupal.md]: ah good catch, I'll add the space before the link

File: docs/drupal/first-deployment-of-drupal.md (hunk 4 above)
Comment (dan2k3k4) [reply docs/drupal/first-deployment-of-drupal.md]: Yeah: https://nvuillam.github.io/mega-linter/descriptors/markdown_markdownlint/#readme - I used the defaults

File: tests/files/drupal8-mariadb-single/web/sites/default/default.settings.php (hunk 8 above)
Comment (dan2k3k4) [reply tests/files/drupal8-mariadb-single/web/sites/default/default.settings.php]: It was already there, I only replaced the `master/slave` instances with `primary/replica` - I _think_ these files are supposed to be the default Drupal provided settings files

File: helpers/check_acme_routes.sh (hunk 7 above)
Comment (dan2k3k4) [reply helpers/check_acme_routes.sh]: Sure

**Review by AlannaBurke — APPROVED** [review]
All of the documentation changes look good to me.

**Pull request merged**